hs_core/hydroshare/utils.py

```diff
@@ -762,13 +762,15 @@ def get_remaining_user_quota(user_or_username, units='MB'):
 
     if user and user.is_active:
         uq = user.quotas.filter(zone='hydroshare').first()
-        if uq:
-            qmsg = QuotaMessage.objects.first()
-            enforce_flag = qmsg.enforce_quota
-            if enforce_flag:
-                remaining = uq.allocated_value - uq.used_value
-                remaining = convert_file_size_to_unit(remaining, to_unit=units, from_unit=uq.unit)
-                return max(remaining, 0)
+        if not uq:
+            # create a quota object for the user
+            uq = user.quotas.create(zone='hydroshare')
+        qmsg = QuotaMessage.objects.first()
+        enforce_flag = qmsg.enforce_quota
+        if enforce_flag:
+            remaining = uq.allocated_value - uq.used_value
+            remaining = convert_file_size_to_unit(remaining, to_unit=units, from_unit=uq.unit) or 0
+            return max(remaining, 0)
     return None
```

Comment:
Why not return zero instead of None so you don't have to handle None on remaining_quota below.

Reply:
we need to be able to destinguish downstream between the cases:
1. That a user's quota doesn't exist or that quota enforcement is not turned on (return None)
2. The user has no remaining quota (return 0)

I will add to the doc string to make this differentiation

Reply:
well actually, I feel like 
```
:return: the remaining quota for the user, or None if the quota is not enforced
```
is adequate...

Reply:
agreed